### DISCUSSION TITLE
Preview of Internal Links

### DISCUSSION THREAD

**Discussion by Soubhik-De** [sumatrapdfreader/sumatrapdf#3326] [Ideas]
Can you add the feature to show a preview of internal links? To clarify: Suppose I make an internal link to equation number 3, so if I take my mouse and hover over this internal link, a small bubble popup window will show me the part of the page which contains the linked equation 3. So that I do not have to click on the link itself to view the linked equation, just the preview would be enough.

This feature is present in the PDF reader Evince in Linux. We'd love to see it here!

**Comment by GitHubRulesOK** [comment]
should already be covered in issues see https://github.com/sumatrapdfreader/sumatrapdf/issues/128, but anyone can try to do a look behind or look ahead window currently there are several ways to attempt that but most have not worked

Easiest current method
From the manual when hyperlinks take you to elsewhere in the document
< Alt > + Left = go back & < Alt >+ Right to go forward (back to where you jumped back from) 😄

related issues https://github.com/sumatrapdfreader/sumatrapdf/issues/1198

**Reply by Soubhik-De** [reply]
Thanks a lot!

**Comment by crystaldot** [comment]
Although it might be complicated to implement this particular feature (#128) but it can immediately boost up the efficiency and workflow of researchers. In my knowledge this feature is already present in pdf reader like - [Evince ](https://wiki.gnome.org/Apps/Evince)and [Sioyek ](https://sioyek.info/). There doesn't exist a single open source pdf reader which supports all the required features altogether that a researcher can use simply and directly.

For examples: 
- Link Previews [ Equations, Citations, Images, Footnotes etc. ]
- Select and Search in web / offline [ Dictionary/Definition, Google (Web search), Google Scholar (supported by [Sioyek ](https://sioyek.info/)) or any other source chosen by the user. ]
- Annotations in any form and saving it either embedded in the pdf or in a different folder (user's choice).
- Import and export of annotations.

**Reply by leanhdung1994** [reply]
You may want to check out https://skim-app.sourceforge.io/.

**Comment by leanhdung1994** [comment]
This option is also available on [Skim](https://skim-app.sourceforge.io/) PDF reader for MacOS. I hope this feature will be implemented.

[![output][1]][1]


  [1]: https://i.sstatic.net/OmtQlP18.png
